### PR TITLE
HW02 Ricardo Muñiz Trejo

### PR DESCRIPTION
**Pull request by rmuniztrejo**
I found this HW very entertaining!

### CODE REVIEW

File: HW02_A_Graph-Fails.Rmd [hunk 1]
```diff
@@ -29,20 +31,33 @@ library("magrittr") #so I can do some piping
...
+\#\# Did you use %>% instead of +?
+
+**The problem in this graph was that we were using** %>% **instead of** `+` **(very self-explanatory) inside the** `ggplot()` **function.**
+
 ```{r}
 data(mpg) #this is a dataset from the ggplot2 package
 
-mpg %>% 
-  ggplot(mapping = aes(x = city, y = hwy, color = "blue")) %>% 
+mpg %>%
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.Rmd:45]: I made the same changes!

File: HW02_A_Graph-Fails.Rmd [hunk 2]
```diff
@@ -29,20 +31,33 @@ library("magrittr") #so I can do some piping
...
+\#\# Error: `mapping` must be created by `aes()`
+\#\# Did you use %>% instead of +?
+
+**The problem in this graph was that we were using** %>% **instead of** `+` **(very self-explanatory) inside the** `ggplot()` **function.**
+
 ```{r}
 data(mpg) #this is a dataset from the ggplot2 package
 
-mpg %>% 
-  ggplot(mapping = aes(x = city, y = hwy, color = "blue")) %>%
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.Rmd:36]: I made the same changes here!

File: HW02_A_Graph-Fails.Rmd [hunk 3]
```diff
@@ -29,20 +31,33 @@ library("magrittr") #so I can do some piping
...
 ### Graph Fail 2
 Why aren't the points blue? It is making me blue that the points in the graph aren't blue :`(
+
+##### Answer 2
+
+**Because blue here is being used inside the** `aes()` **function, R interpreted it as a variable of the graph (which is why appears inside the legend), but "blue" is not a vector within mpg, so it cannot be shown in the plot. Instead, the** `color = "blue"` **argument should be specified outside**  `aes`**.**
+
 ```{r}
 ggplot(data = mpg) + 
-  geom_point(mapping = aes(x = displ, y = hwy, color = "blue"))
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.Rmd:45]: Points are blue now. Good job :)

File: HW02_A_Graph-Fails.Rmd [hunk 4]
```diff
@@ -52,35 +67,63 @@ ggplot(data = mpg) +
 Two mistakes in this graph. First, I wanted to make the the points slightly bolder, but changing the alpha to 2 does nothing. What does alpha do and what does setting it to 2 do? What could be done instead if I want the points slightly bigger? 
 
 Second, I wanted to move the legend on top of the graph since there aren't any points there, putting it at approximately the point/ordered pair (5, 40). How do you actually do this? Also, how do you remove the legend title ("class")? Finally, how would you remove the plot legend completely?
+
+##### Answer 3
+
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.Rmd:72]: Very thorough explanations on the changes made and the appropriate use of some arguments.

File: HW02_A_Graph-Fails.Rmd [hunk 5]
```diff
@@ -52,35 +67,63 @@ ggplot(data = mpg) +
...
+**Lastly, to remove the legend completely, you should write** `legend.position = "none"`**.**
+
+
 ```{r}
 mpg %>% 
 ggplot() + 
-  geom_point(mapping = aes(x = displ, y = hwy, color = class), alpha = 2) + 
-  theme(legend.direction = "horizontal") + 
-  theme(legend.position = c(5, 40))
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.Rmd:60]: The Figure looks as requested! Also, changing the legend position to "horizontal" was a good idea. I did not do this and the Figure looks way better this way.

File: HW02_A_Graph-Fails.Rmd [hunk 6]
```diff
@@ -52,35 +67,63 @@ ggplot(data = mpg) +
...
+
+**The problem here was that we were using drv as a color aesthetic, so ggplot was interpreting every color as a different set of data, not just one, and because of that several smooth lines were drawn. To correct it, you must quit drv from the** `aes()` **function.**
+
+**Now, if you want to keep the color scheme according to drv without losing the unique smooth line, you can add the color as an exclusive attribute of** `geom_point()` **as shown in the code.**
+
 ```{r}
 mpg %>% 
-ggplot(mapping = aes(x = displ, y = hwy, color = drv)) + 
-  geom_point() + 
-  geom_smooth(se = F) #se = F makes it so it won't show the error in the line of fit
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.Rmd:71]: This was a good strategy. Simple and clearly explained (and it worked, of course).

File: HW02_A_Graph-Fails.Rmd [hunk 7]
```diff
@@ -52,35 +67,63 @@ ggplot(data = mpg) +
...
+ggplot(mapping = aes(x = displ, y = hwy)) + 
+  geom_point(aes(color = drv)) + 
+  geom_smooth(se = F) #se = F makes it so it won't show the error in the line of fit +
 ```
 
 ### Graph Fail 5
 I got tired of the points, so I went to boxplots instead. However, I wanted the boxes to be all one color, but setting the color aesthetic just changed the outline? How can I make the box one color, not just the outline?
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.Rmd:112]: To be honest I am slightly uncertain of what was expected here. I for one, did not remove the "colo" argument. I left both "fill" and "color" there. However, the change seems appropriate given what was requested.

File: HW02_A_Graph-Fails.Rmd [hunk 8]
```diff
@@ -52,35 +67,63 @@ ggplot(data = mpg) +
...
 I got tired of the points, so I went to boxplots instead. However, I wanted the boxes to be all one color, but setting the color aesthetic just changed the outline? How can I make the box one color, not just the outline? 
 
-Also, the x-axis labels were overlaping, so I rotated them. But now they overlap the bottom of the graph. How can I fix this so axis labels aren't on the graph?
+Also, the x-axis labels were overlapping, so I rotated them. But now they overlap the bottom of the graph. How can I fix this so axis labels aren't on the graph?
+
+##### Answer 5
+
+`Color ` **argument only affects the outline of the figure, you need the** `fill` **argument to add color to the entire box.**
 
+**To be honest, I'm not sure if there's a better solution, but one way to avoid overlapping is by increasing the length of the ticks to move the labels below the graph lower limit.**
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.Rmd:121]: To be honest, I am not sure either. But this seems to work fine to me!

File: HW02_A_Graph-Fails.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
+What went wrong?
```
Comment (lcespedesarias) [inline HW02_A_Graph-Fails.md:1]: I added the comments in the .Rmd

File: HW02_B_Mimic_starter.Rmd [hunk 10]
```diff
@@ -36,22 +36,48 @@ Try for all 4, but if you are struggling don't worry about it. Try your best for
 ```{r, echo = T, include = TRUE}
 data("diamonds")
 #hint think about the *position* the bars are in...
+
 ```
 
 
 Using the diamonds dataset, make this graph:
-```{r graph1 code, echo=FALSE}
-
```
Comment (lcespedesarias) [inline HW02_B_Mimic_starter.Rmd:44]: The figure looks great. Exactly as it was requested. And I learned how to add the gray shading, because I was not able to do that successfully. The annotations within the code are useful to understand it better!

File: HW02_B_Mimic_starter.Rmd [hunk 11]
```diff
@@ -36,22 +36,48 @@ Try for all 4, but if you are struggling don't worry about it. Try your best for
...
 
 ### Graph 2
 ```{r, echo = T, include = TRUE}
 data("iris")
+
+
 ```
 
 Using the iris dataset, make this graph:
-```{r graph 2 code, echo=FALSE}
```
Comment (lcespedesarias) [inline HW02_B_Mimic_starter.Rmd:54]: The re-ordering strategy was very clever (by creating a new column within the data table). I struggle to get both the order and colors right, so I ended up leaving them in the wrong order. Something you can do to make the code simpler is to change the value of the "scales" argument within facet_wrap to "free_y". That way you don't have to resize the x axis after.

File: HW02_B_Mimic_starter.Rmd [hunk 12]
```diff
@@ -64,11 +90,26 @@ corvette <- mpg[mpg$model == "corvette",]
 #install
 require("ggrepel") #useful for making text annotations better, hint hint
 set.seed(42)
+
 ```
 
 Now using the mpg dataset and the corvette dataset, make this graph:
 
-```{r graoh 3 code}
```
Comment (lcespedesarias) [inline HW02_B_Mimic_starter.Rmd:71]: The figure looks great. Just as requested. I found the code for this figure a little hard to follow, but it works well! Particularly, I was confused about the "label" argument within "ggplot", perhaps adding it within "geom_text_repel" would be a bit more clear (this is the function that actually add the label, but the labels themselves are specified outside). This is totally personal though (I might be even saying something wrong too!), as it works perfectly. 

The strategy for coloring the corvette points in a different color (blue) was very clever. I just added blue points on top (for corvettes) of the black ones (for all cars), which is not a great solution. Other thing I learned!

File: HW02_B_Mimic_starter.Rmd [hunk 13]
```diff
@@ -82,15 +123,24 @@ data(mpg)
...
 display.brewer.all(colorblindFriendly = T) #take a look at the colorblindfriendly options
+
 ```
 
-The above graph lets you see some colobrlind friendly palettes. For the graph below, I used Set2.
+The above graph lets you see some colorblind friendly palettes. For the graph below, I used Set2.
 
 
 
 Now using the above mpg dataset, make this graph
```
Comment (lcespedesarias) [inline HW02_B_Mimic_starter.Rmd:134]: The figure looks great. The only thing I would change is the position of the points relative to the outiline of the boxplots. Here points are over the boxes, and they should be below. For that, you would need to 1) Change the order of "geom_boxplot" and "geom_jitter" by putting "geom_jitter" first 2) Add the argument "fill=NA" to "geom_boxplot" so the boxes are not filled and you can actually see the points that are below.

File: HW02_B_Mimic_starter.md [hunk 14]
```diff
@@ -0,0 +1,159 @@
+HW02\_B\_Graph-Mimic
```
Comment (lcespedesarias) [inline HW02_B_Mimic_starter.md:1]: I added comments in the .Rmd

**Review by lcespedesarias — COMMENTED** [review]
Overall the coding strategy and presentation was excellent. There were some clever solutions to problems that I did not thought of before. The reproducibility was also excellent, as I was able to replicate he figures very easily. I only found one difference between the figures produced with this code and the figures that were requested in the homework. Great work 👍

File: HW02_A_Graph-Fails.Rmd (hunk 8 above)
Comment (rmuniztrejo) [reply HW02_A_Graph-Fails.Rmd:121]: I saw your script; I think your solution was more elegant.

File: HW02_B_Mimic_starter.Rmd (hunk 11 above)
Comment (rmuniztrejo) [reply HW02_B_Mimic_starter.Rmd:54]: It took me a while to get this graph done. I had to check the documentation, look for examples, and experimenting a lot with some functions. It was not easy. I didn't know about the "free_y" argument, its way more useful, thank you!
